### PR TITLE
[RFR] change board representation

### PR DESCRIPTION
**Pull request by mchaffotte**
- adding a fence, remove an edge in the model
- add a form in each border of each square to add a fence

AddFence

### CODE REVIEW

**Comment by fzaninotto** [comment]
Can you please add a screenshot?

**Comment by mchaffotte** [comment]
I change it internally, I use right now a graph from a java library.
I will add a screenshot of the feature when the front will be (almost) done

**Comment by mchaffotte** [comment]
@fzaninotto this is the preview of adding fences

File: src/main/java/com/marmelab/quoridor/web/AddFence.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import com.marmelab.quoridor.model.Position;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class AddFence {
+
+    private final List<Fence> addHorizontalFences;
```
Comment (floo51) [inline src/main/java/com/marmelab/quoridor/web/AddFence.java]: This is named as a method but it isn't.

File: src/main/java/com/marmelab/quoridor/web/AddFence.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    private final List<Fence> addVerticalFences;
+
+    public AddFence(int boardSize, List<Fence> horizontalFences, List<Fence> verticalFences) {
```
Comment (floo51) [inline src/main/java/com/marmelab/quoridor/web/AddFence.java]: This method contains too much logic at once. You maybe could put into named methods whatever it does inside the for loops

File: src/main/java/com/marmelab/quoridor/web/BoardView.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+        final List<Fence> fences = game.getBoard().getFences();
+
+        verticalFences = fences.stream()
+                .filter(fence -> !fence.isHorizontal())
+                .collect(Collectors.toList());
+        horizontalFences = fences.stream()
+                .filter(Fence::isHorizontal)
+                .collect(Collectors.toList());
+        AddFence addFence = new AddFence(game.getBoard().getBoardSize(), horizontalFences, verticalFences);
```
Comment (floo51) [inline src/main/java/com/marmelab/quoridor/web/BoardView.java]: This is named as a method and isn't.

File: src/test/java/com/marmelab/quoridor/web/QuoridorEndpointTest.java [hunk 4]
```diff
@@ -32,8 +32,11 @@ public void gameWithBoardAndPawn() throws Exception {
 
         String content = quoridor.getResponse().getContentAsString();
         assertThat(content)
-                .contains("<div class=\"pawn\" style=\"top: 160px; left: 0px\">")
-                .contains("<div class=\"square\" style=\"top: 0px; left: 80px\"></div>");
+                .contains(" <div class=\"pawn\"\n" +
```
Comment (floo51) [inline src/test/java/com/marmelab/quoridor/web/QuoridorEndpointTest.java]: Testing the html result probably isn't the best choice. This gonna break every time you touch templates. You better just test what it says it tests : the presence of the pawn.

**Review by floo51 — CHANGES_REQUESTED** [review]

File: src/test/java/com/marmelab/quoridor/model/BoardTest.java [hunk 5]
```diff
@@ -160,8 +158,7 @@ void addFenceNotPossibleHorizontalFenceOnVerticalFence() {
     }
 
     @Test
-    @DisplayName("It should not be possible to add a vertical fence before a vertical fence")
-    @Disabled
+    @DisplayName("It should not be possible to add an vertical fence before a vertical fence")
```
Comment (floo51) [inline src/test/java/com/marmelab/quoridor/model/BoardTest.java]: s/an/a/

**Review by floo51 — CHANGES_REQUESTED** [review]

**Review by floo51 — APPROVED** [review]

**Pull request merged**